crates/ruma-events/src/matrix_rtc/member_event.rs

```diff
@@ -0,0 +1,234 @@
+//! Types for matrixRTC state events ([MSC3401]).
+//!
+//! This implements a newer/updated version of MSC3401.
+//!
+//! [MSC3401]: https://github.com/matrix-org/matrix-spec-proposals/pull/3401
+
+use ruma_common::{MilliSecondsSinceUnixEpoch, OwnedUserId};
+use ruma_macros::EventContent;
+use serde::{Deserialize, Serialize};
+
+use super::{
+    focus::{ActiveFocus, Focus},
+    member_data::{Application, LegacyMembershipData, MembershipData, SessionMembershipData},
+};
+use crate::{PossiblyRedactedStateEventContent, StateEventType};
+
+/// The member state event for a matrixRTC session.
+///
+/// This is the object containing all the data related to a matrix users participation in a
+/// matrixRTC session.
+///
+/// This is a unit struct with the enum [`MemberEventContent`] because a Ruma state event cannot be
+/// an enum and we need this to be an untagged enum for parsing purposes. (see
+/// [`MemberEventContent`])
+///
+/// This struct also exposes allows to call the methods from [`MemberEventContent`].
+#[derive(Clone, Debug, Serialize, Deserialize, EventContent)]
+#[ruma_event(type = "org.matrix.msc3401.call.member", kind = State, state_key_type = OwnedUserId, custom_possibly_redacted)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct CallMemberEventContent(pub MemberEventContent);
```

Comment:
Is there a reason this is not directly an enum?

Reply:
yes, enums are not supported by the `EventContent` macro. (I could not get them to work with it at least. Is there a way? That would be good)

Reply:
It works, you just need to use the `custom_redacted` attribute and implement `RedactedCallMemberEventContent` yourself.

Reply:
Its very nice that this works. Thanks.